pkg/storage/unified/sql/backend.go

```diff
@@ -631,6 +632,79 @@ func (b *backend) listLatest(ctx context.Context, req *resourcepb.ListRequest, c
 	return iter.listRV, err
 }
 
+// ListModifiedSince will return all resources that have changed since the given resource version.
+// If a resource has changes, only the latest change will be returned.
+func (b *backend) ListModifiedSince(ctx context.Context, key resource.NamespacedResource, sinceRv int64) (int64, iter.Seq2[*resource.ModifiedResource, error]) {
+	var latestRv int64
+	err := b.db.WithTx(ctx, ReadCommittedRO, func(ctx context.Context, tx db.Tx) error {
+		var err error
+		latestRv, err = b.fetchLatestRV(ctx, tx, b.dialect, key.Group, key.Resource)
+		return err
+	})
+	if err != nil {
+		return 0, func(yield func(*resource.ModifiedResource, error) bool) {
+			yield(nil, err)
+		}
+	}
+
+	// track seen resources to avoid duplicates
+	seen := make(map[string]bool)
```

Comment:
You can change this to `struct{}` so you don't waste any memory on this.
```suggestion
	seen := make(map[string]struct{})
```

Reply:
Adding to that: given the SQL is sorted by `(group ASC, resource ASC, name ASC, rv DESC)`, we don't need to remember all the resources seen previously, but only the last one. 
```
     if last_seen == mr.Key.Name { continue}
     last_seen = mr.Key.Name 
```
Could also consider a small check to ensure someone doesn't change the sort order in the future 🤕
```
   // 
   if mr.Key.Name <= last_seen { // should never happen, return an error ! }
```